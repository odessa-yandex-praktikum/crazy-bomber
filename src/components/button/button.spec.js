import React from 'react';
import renderer from 'react-test-renderer';
import {Button, EButtonColor, EButtonType} from './index';
import {configure, shallow} from 'enzyme';
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';

configure({adapter: new Adapter()});

it('renders correctly Ok', () => {
    const tree = renderer
        .create(
            <Button
                key={1}
                text="OK"
                buttonColor={EButtonColor.PRIMARY}
                buttonType={EButtonType.FORM}
                onClick={console.log('click')}
            />
        )
        .toJSON();
    expect(tree).toMatchSnapshot();
});

it('renders correctly Error', () => {
    const tree = renderer
        .create(
            <Button
                key={2}
                text="Error"
                buttonColor={EButtonColor.ERROR}
                buttonType={EButtonType.ERROR}
                onClick={console.log('click error')}
            />
        )
        .toJSON();
    expect(tree).toMatchSnapshot();
});

it('renders correctly full screen', () => {
    const tree = renderer
        .create(
            <Button
                key={4}
                text="Full screen"
                buttonColor={EButtonColor.PRIMARY}
                buttonType={EButtonType.FULL_SCREEN}
                onClick={console.log('click full screen')}
            />
        )
        .toJSON();

    expect(tree).toMatchSnapshot();
});

it('renders correctly exit full screen', () => {
    const tree = renderer
        .create(
            <Button
                key={5}
                text="Exit full screen"
                buttonColor={EButtonColor.PRIMARY}
                buttonType={EButtonType.EXIT_FULL_SCREEN}
                onClick={console.log('click exit full screen')}
            />
        )
        .toJSON();
    expect(tree).toMatchSnapshot();
});

it('clicks correctly', () => {
    const mockCallBack = jest.fn();
    const button = shallow(
        <Button
            key={6}
            text="OK"
            buttonColor={EButtonColor.PRIMARY}
            buttonType={EButtonType.FORM}
            onClick={mockCallBack}
        />
    );
    button.find('button').simulate('click');
    button.find('button').simulate('click');
    expect(mockCallBack.mock.calls.length).toEqual(2);
});
