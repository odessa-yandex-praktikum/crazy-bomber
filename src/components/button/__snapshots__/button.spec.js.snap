// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly Error 1`] = `
<button
  className="button color-error button__type-error "
  type="button"
>
  Error
</button>
`;

exports[`renders correctly Ok 1`] = `
<button
  className="button color-primary button__type-form "
  type="button"
>
  OK
</button>
`;

exports[`renders correctly exit full screen 1`] = `
<button
  className="button color-primary button__exit-full-screen "
  type="button"
>
  Exit full screen
</button>
`;

exports[`renders correctly forum 1`] = `
<button
  className="button color-secondary forum-page__button "
  type="button"
>
  forum
</button>
`;

exports[`renders correctly full screen 1`] = `
<button
  className="button color-primary button__full-screen "
  type="button"
>
  Full screen
</button>
`;
